src/experimental/Db.cpp

```diff
@@ -0,0 +1,42 @@
+#include "Db.h"
+
+using namespace experimental;
+
+bool DB::init()
+{
+    const QString driver("QSQLITE");
+
+    if (QSqlDatabase::isDriverAvailable(driver)) {
+
+        QDir directory("data");
+        QString databaseLocation = directory.filePath("logbook.db");
+        QSqlDatabase db = QSqlDatabase::addDatabase(driver);
+        db.setDatabaseName(databaseLocation);
+
+        if (!db.open()) {
+            DEB("DatabaseConnect - ERROR: " << db.lastError().text());
+            return false;
+        } else {
+            DEB("Database connection established.");
+            // Enable foreign key restrictions
+            QSqlQuery query("PRAGMA foreign_keys = ON;");
+            // Retreive database layout and store in member variables
+            tables = db.tables();
+
+            QStringList columnNames;
+            for (const auto &table : tables) {
+                columnNames.clear();
+                QSqlRecord fields = db.record(table);
+                for (int i = 0; i < fields.count(); i++) {
+                    columnNames.append(fields.field(i).name());
+                    database_layout.insert(table, columnNames);
+                    ///[F] database_layout.value("pilots") gives you the columns for the pilots table
+                }
+            }
+            return true;
+        }
+    } else {
+        DEB("DatabaseConnect - ERROR: no driver " << driver << " available");
+        return false;
+    }
+}
```

Comment:
Broader programming comment. There is a specific way of handling exiting state in such if/else type of logic. in order to avoid nesting if and indenting you put the failure state first as follows
```c++
if(!desirable_state) return false;
// continuing normally
if(!second_desirable_statate) return true;
// more logic
// FINALLY
return true
```
So you make your logic more linear with exit hatches as it goes further down. 
For our example it could be.
```c++
if(!availlableDriver()) return false;
auto db = QSqlDatabase(...);
if(!db.open()) return false;
setupTables(db);
return true;
```
That way the flow is up down linearly with no indentation